spicepy/_http.py

```diff
@@ -20,21 +19,28 @@ def __init__(self, base_url: str, headers: Dict[str, str]) -> None:
 
         self.base_url = base_url
 
+    # pylint: disable=R0913
+    # pylint: disable=R0917
     def send_request(
         self,
         method: HttpMethod,
         path: str,
         param: Optional[Dict[str, Any]] = None,
-        body: Optional[Union[Any, bytes, str]] = None,
+        headers: Optional[Dict[str, Any]] = None,
+        body: Optional[str] = None,
     ) -> Any:
-        if not isinstance(body, (bytes, str)) and body is not None:
-            body = json.dumps(body)
+        if headers is None:
+            headers = {}
+            headers.update(self.session.headers)
```

Comment:
Should we do this here and [L36](https://github.com/spiceai/spicepy/pull/119/files#diff-55e83810c5474c3c7788d48433692743323d34c1783f00fc95c990805d7f7babR36).